extensions/resteasy-classic/rest-client/runtime/src/main/java/io/quarkus/restclient/runtime/RestClientBase.java

```diff
@@ -322,10 +322,13 @@ private RestClientConfig clientConfigByClassName() {
         return this.configRoot.getClientConfig(this.proxyType);
     }
 
-    private static <T> Optional<T> oneOf(Optional<T> o1, Optional<T> o2) {
-        if (o1.isPresent()) {
-            return o1;
+    @SafeVarargs
+    private static <T> Optional<T> oneOf(Optional<T>... optionals) {
```

Comment:
If we assume that the global values don't need to be `Optional`, then we just use `orElse` if the specific client property is not available.